### DISCUSSION TITLE
Fixed app crashing but question persists what happened

### DISCUSSION THREAD

**Discussion by hellogbg** [invertase/react-native-firebase#4480] [Q&A]
I managed to fix my app that was constantly crashing after updating React Native to latest version. I updated fnfirebase components as well. 
I wasn't getting any error messages in the terminal window nor in Chrome, but after a while I discovered this error in Xcode when running the app from there: 
"The Google Mobile Ads SDK was initialized without an application ID. Google AdMob publishers, follow instructions at https://googlemobileadssdk.page.link/admob-ios-update-plist to set a valid application ID. Google Ad Manager publishers, follow instructions at https://googlemobileadssdk.page.link/ad-manager-ios-update-plist".

If I would clean cache, clean build folder, clean node_modules, or pod install, then app would work only first time i run it, but not second time or any other time after that.

The only thing that worked was to add this code to info.plist:
```
<key>GADIsAdManagerApp</key>
<true/>
<key>SKAdNetworkItems</key>
  <array>
    <dict>
      <key>SKAdNetworkIdentifier</key>
      <string>cstr6suwn9.skadnetwork</string>
    </dict>
  </array>
```

as stated here: https://developers.google.com/ad-manager/mobile-ads-sdk/ios/quick-start#update_your_infoplist

There is nothing about this in rnfirebase docs and I would like to find out what I was/am doing wrong? Is this just missing in the docs or do I have some code left from old versions of rnfirebase?

I am using:
"@react-native-firebase/admob": "^7.6.10",
"@react-native-firebase/app": "^8.4.7",

"react-native": "0.63.3",

P.s. I use rnfirebase in other apps where I don't use AdMob and everything works without editing info.plist

**Comment by mikehardy** [comment]
I think this is what you are looking for, in the install docs

https://rnfirebase.io/admob/usage#setting-up-google-admob

**Reply by hellogbg** [reply]
I have set everything up in the older versions of @react-native-firebase/admob and it worked before. 
I am using: <project-root>/firebase.json since before and I have apps setup in google admob account.
I just didn't have configured outbound requests, but now i did in useEffect and when I try to remove the code in info.plist i am getting same behaviour as before, app crash. (I have updated the info.plist code in my original question, some lines disappeared when i didn't use code block)

**Reply by mikehardy** [reply]
Can't reproduce, this thing does it all full auto and works: https://github.com/mikehardy/rnfbdemo/blob/master/make-demo.sh

Specifically https://github.com/mikehardy/rnfbdemo/blob/fc7e943ca95fe1777300985f31db2a075d2da4a5/make-demo.sh#L132-L156

Something is wrong with your project, or perhaps you haven't rebuilt it to render the firebase.json into the plist

**Reply by hellogbg** [reply]
I assume that something happened when upgrading React Native, and I don't know what, and I would like to find actual reason for this happening. 
How do I "rebuild it to render the firebase.json into the plist"?

This is my Podfile:

```require_relative '../node_modules/react-native/scripts/react_native_pods'
require_relative '../node_modules/@react-native-community/cli-platform-ios/native_modules'

platform :ios, '10.0'

target 'XXXXXXXX' do
  config = use_native_modules!

  use_react_native!(:path => config["reactNativePath"])

  target 'XXXXXXXX' do
    inherit! :complete
    # Pods for testing
  end

  # Enables Flipper.
  #
  # Note that if you have use_frameworks! enabled, Flipper will not work and
  # you should disable these next few lines.
  use_flipper!
  post_install do |installer|
    flipper_post_install(installer)
  end
end

target 'XXXXXXXX-tvOS' do
  # Pods for XXXXXXXX-tvOS

  target 'XXXXXXXX-tvOSTests' do
    inherit! :search_paths
    # Pods for testing
  end

end
```

**Reply by mikehardy** [reply]
`npx react-native-clean-project` to make sure you cleaned out your old ios build, Then rebuild your ios app

**Reply by hellogbg** [reply]
I have tried that, no success.
Somehow my project doesn't fetch app id from firebase.json
If I uninstall @react-native-firebase/admob then everything works, but not if I install it again. Do you have any other tips on what I can try?

p.s. Has it something to do with this https://stackoverflow.com/a/58173681/9123009
AdMob vs Ad Manager. Is there any difference regarding that and rnfirbase?
I use some of my own ads but it works on android and worked great before on iOs as well.

**Reply by mikehardy** [reply]
Run my script. Prove to yourself it works?

**Reply by hellogbg** [reply]
Don't misunderstand me, I don't doubt that your script works (and component as well). I know that something is wrong on my end and I am just trying to figure out what it is and how I can solve it without adding anything to info.plist as I know it can bring me issues with future upgrades

**Reply by hellogbg** [reply]
I have been working with this issue for countless hours now :) Just wanted you to know about my findings and for others that may be interested.
Running in Xcode I could see that importing GADApplicationIdentifier (https://developers.google.com/admob/ios/quick-start#update_your_infoplist) didn't work. It didn't get imported to Info.plist simply. Console message from your script was that the GADApplicationIdentifier was already in Info.plist but it wasn't. 

I also tried to setup a new project with using only rnfirebase app and AdMob but I got the same error.

I also found that some error message that I came across had to do with a new feature in Xcode and then I started to suspect that it can be something with Xcode 12.1 (and 12) that I was using.

So my final solution (that works for now) was to downgrade to Xcode 11.5 and now everything works as expected.

**Reply by mikehardy** [reply]
@avenyproduction that is most strange. The problem with downgrade to Xcode 12 is that you won't be able to use the new APIs for IDFA consent required by apple on ios14 starting in a few months. I wish I had a better answer, but I'm not sure what is going on - I just know that building with non-current Xcode is generally not supported as it blocks adoption of new APIs

**Reply by hellogbg** [reply]
I was thinking of upgrading xcode later, but as my computer with the code was stolen, I can give Xcode 12 another chance when i try ti re-implement the upgrade on a new computer

**Reply by mikehardy** [reply]
Oh that's terrible. My condolences @avenyproduction
